File: _depot/scripts/_rb/report.rb
# exports: Table, List
# ring: 0 (PURE)
# depends-on: ./loader

Table = ->(rows, headers) {
  widths = headers.map.with_index { |h, i| [h.size, *rows.map { |r| r[i].to_s.size }].max }
  head = headers.map.with_index { |h, i| h.ljust(widths[i]) }.join(" | ")
  sep = widths.map { |w| "-" * w }.join("-|-")
  body = rows.map { |r| r.map.with_index { |v, i| v.to_s.ljust(widths[i]) }.join(" | ") }.join("\n")
  "#{head}\n#{sep}\n#{body}"
}

List = ->(items) { items.map { |i| "- #{i}" }.join("\n") }

File: .opencode/_scripts/_golib/internal/report/report.go
// Package report renders aligned tables and bullet lists.
// ring: 0 (PURE) — no I/O.
package report

import (
	"fmt"
	"strings"
)

// FormatTable renders rows into aligned columns with a header and separator.
func FormatTable(rows [][]string, headers []string) string {
	if len(rows) == 0 {
		return "(empty)"
	}
	widths := make([]int, len(headers))
	for i, h := range headers {
		widths[i] = len(h)
	}
	for _, row := range rows {
		for i, val := range row {
			if i < len(widths) && len(val) > widths[i] {
				widths[i] = len(val)
			}
		}
	}
	pad := func(cells []string) string {
		parts := make([]string, len(cells))
		for i, cell := range cells {
			parts[i] = fmt.Sprintf("%-*s", widths[i], cell)
		}
		return strings.Join(parts, " | ")
	}
	head := make([]string, len(headers))
	for i, h := range headers {
		head[i] = fmt.Sprintf("%-*s", widths[i], h)
	}
	seps := make([]string, len(widths))
	for i, w := range widths {
		seps[i] = strings.Repeat("-", w)
	}
	var body []string
	for _, row := range rows {
		body = append(body, pad(row))
	}
	return strings.Join(append([]string{strings.Join(head, " | "), strings.Join(seps, "-|-")}, body...), "\n")
}

// FormatList renders items as a bullet list.
func FormatList(items []string) string {
	if len(items) == 0 {
		return ""
	}
	lines := make([]string, len(items))
	for i, item := range items {
		lines[i] = "- " + item
	}
	return strings.Join(lines, "\n")
}
